### PR TITLE
chore: cleanup the `R` folder

### PR DESCRIPTION
**Pull request by etiennebacher**
* prefix files related to I/O with `io_` so that they're easier to find in the folder
* remove useless docs for `.DollarNames`
* remove a couple of unused functions
* move code from `rstudio_completion.R` to `autocompletion.R` 
* move code from `namespace.R` to `polars-package.R`
* move code from `rlang.R` to `utils.R`

### CODE REVIEW

**Comment by etiennebacher** [comment]
@eitsupi this will likely create conflicts with later PRs, so I'd like to merge it as early as possible. Are you ok with those changes?

File: R/utils.R [hunk 1]
```diff
@@ -637,3 +637,40 @@ raw_rownames = function(x) {
...
+
+
+# Adapted (or copied) functions from {rlang}
+`%||%` = function(x, y) {
+  if (is.null(x)) y else x
+}
+
+`%|||%` = function(x, y) {
+  if (!length(x)) y else x
+}
```
Comment (eitsupi) [inline R/utils.R:649]:
```suggestion
```

This can be removed.

File: R/polars-package.R [hunk 2]
```diff
@@ -4,3 +4,8 @@
 ## usethis namespace: start
 ## usethis namespace: end
 NULL
+
+#' @importFrom utils globalVariables head tail download.file capture.output str
+NULL
```
Comment (eitsupi) [inline R/polars-package.R:9]: I believe this is generally placed on the upper block.
As follows.

```r
## usethis namespace: start
#' @importFrom utils globalVariables head tail download.file capture.output str
## usethis namespace: end
NULL
```

**Review by eitsupi — APPROVED** [review]
Thanks, a few comments.

**Pull request merged**